### DISCUSSION TITLE
Identifying C-MOVE requests in SCP C-STORE

### DISCUSSION THREAD

**Discussion by andyle2k** [fo-dicom/fo-dicom#1683] [Q&A]
Hi, 

Basically, is there any way for me to identify a C-MOVE that I've initiated when it comes back to my SCP as a C-STORE? Or potentially add an identifier of some sort when I initiate the C-MOVE and have that identifier sent to the SCP during the C-STORE?

Example:

I wish to download a study temporarily and store it in a unique directory so I can safely delete the directory after I'm done with it. Currently it is downloaded to ``[root]/[study_uid]/[series_uid]/[instance_uid].dcm``
Most of the time that is fine but if another user needs to download the same study/series/instance while the first thread is still working on the files you get clashes:
1. First thread might lock the DICOM files
2. First thread might be done before second thread and have deleted the directory

Thanks.

**Comment by knopkem** [comment]
Hi,
this post might help you: https://stackoverflow.com/questions/14259852/how-to-identify-image-receive-in-c-store-as-result-of-a-c-move-query
Other than that you can always use some form of locking (this might be harder with a threaded server though), I did this on the node proxy, see https://github.com/knopkem/dicomweb-proxy/blob/master/src/dimse/fetchData.ts#L27
Cheers
